webapp/loggedin/surveyViews/nodeDefEdit/basic/basicProps.js

```diff
@@ -52,13 +28,41 @@ const BasicProps = props => {
   } = props
   const validation = getValidation(nodeDef)
 
+  const { i18n } = useContext(AppContext)
+
+  const displayAsItems = [
+    {
+      key: NodeDefLayout.nodeDefRenderType.form,
+      label: i18n.t('nodeDefEdit.basicProps.form'),
+    },
+    {
+      key: NodeDefLayout.nodeDefRenderType.table,
+      label: i18n.t('nodeDefEdit.basicProps.table'),
+    },
+  ]
+
+  const displayInItems = [
+    {
+      key: NodeDefLayout.nodeDefDisplayIn.parentPage,
+      label: i18n.t('nodeDefEdit.basicProps.parentPage'),
+    },
+    {
+      key: NodeDefLayout.nodeDefDisplayIn.ownPage,
+      label: i18n.t('nodeDefEdit.basicProps.ownPage'),
+    },
+  ]
+
+  const onPropLabelsChange = (labelItem, key, currentValue) => {
+    putNodeDefProp(nodeDef, key, R.assoc(labelItem.lang, labelItem.label, currentValue))
+  }
+
   return (
```

Comment:
since displayAsItems and displayInItems are in a way not part of the component, i would have left them outside and converted them as a method . e.g. ```getDisplayInItems = i18n => ([..., ...])``` what do you think?